File: EcuacionLineal_Java/EcuacionLineal.java
public class EcuacionLineal {

    public float coeficienteA;
    public float coeficienteB;
    public float terminoIndependienteC;
    public float coeficienteD;
    public float coeficienteE;
    public float terminoIndependienteF;
    public float determinante;
    public float variableX;
    public float variableY;

    public EcuacionLineal(float coeficienteA, float coeficienteB, float terminoIndependienteC, float coeficienteD, float coeficienteE, float terminoIndependienteF) {
        this.coeficienteA = coeficienteA;
        this.coeficienteB = coeficienteB;
        this.terminoIndependienteC = terminoIndependienteC;
        this.coeficienteD = coeficienteD;
        this.coeficienteE = coeficienteE;
        this.terminoIndependienteF = terminoIndependienteF;
        this.determinante = this.coeficienteA * this.coeficienteE - this.coeficienteB * this.coeficienteD;
        this.variableX = this.coeficienteE * this.terminoIndependienteC - this.coeficienteB * this.terminoIndependienteF / this.determinante;
        this.variableY = this.coeficienteA * this.terminoIndependienteF - this.coeficienteD * this.terminoIndependienteC / this.determinante;
    }

    public float getDeterminante() {
        return determinante;
    }

    public float getVariableX() {
        return variableX;
    }

    public float getVariableY() {
        return variableY;
    }

    public static void main(String[] args){
        EcuacionLineal ecuacionLineal = new EcuacionLineal(2, 1, 3, 4, 8, 6);
        System.out.println(ecuacionLineal.getDeterminante());
        System.out.println(ecuacionLineal.getVariableX());
        System.out.println(ecuacionLineal.getVariableY());
    }
}

File: EcuacionLineal_Python/Metodos/ecuacionLineal.py
class EcuacionLineal:
    __coeficienteA = float(0)
    __coeficienteB = float(0)
    __terminoIndependienteC = float(0)
    __coeficienteD = float(0)
    __coeficienteE = float(0)
    __terminoIndependienteF = float(0)
    __determinante = float(0)
    __variableX = float(0)
    __variableY = float(0)


    #Metodo constructor
    def __init__(self, coeficienteA, coeficienteB, terminoIndependienteC, coeficienteD, coeficienteE, terminoIndependienteF):
        self.__coeficienteA = coeficienteA
        self.__coeficienteB = coeficienteB
        self.__terminoIndependienteC = terminoIndependienteC
        self.__coeficienteD = coeficienteD
        self.__coeficienteE = coeficienteE
        self.__terminoIndependienteF = terminoIndependienteF
        self.calcularDeterminante()
        self.calcularVariableX()
        self.calcularVariableY()

    #Metodo destructor
    def __del__(self):
        self.calcularDeterminante()
        self.calcularVariableX()
        self.calcularVariableY()

    # Sobrecarga de operadores
    def __str__(self):
        return "la variable X de la escuacion es" + self.__VariableX + "mientras que la variable Y es" + self.__variableY


    #Metodo de clase
    @classmethod
    def calcularDeterminante(cls, coeficienteA,  coeficienteE, coeficienteB, coeficienteD):
        determinante = coeficienteA * coeficienteE - coeficienteB * coeficienteD
        return determinante

    # Metodo de instancia
    def calcularVariableX(self):
        self.__VariableX = self.__coeficienteE * self.__terminoIndependienteC - self.__coeficienteB * self.__terminoIndependienteF / self.__determinante
    def get_variableX(self):
        return self.__variableX

    def calcularVariableY(self):
        self.__variableY = self.__coeficienteA * self.__terminoIndependienteF - self.__coeficienteD * self.__terminoIndependienteC / self.__determinante
    def get_variableY(self):
       return self.__variableY




